executor/common_usb.h

```diff
@@ -456,6 +458,48 @@ static bool lookup_control_io_response(struct vusb_descriptors* descs, struct vu
 	return false;
 }
 
+#if USB_DEBUG
+#include <linux/hid.h>
+#include <linux/usb/cdc.h>
+#include <linux/usb/ch11.h>
+#include <linux/usb/ch9.h>
+
+static void analyze_control_request(struct usb_ctrlrequest* ctrl)
+{
+	switch (ctrl->bRequestType & USB_TYPE_MASK) {
+	case USB_TYPE_STANDARD:
+		switch (ctrl->bRequest) {
+		case USB_REQ_GET_DESCRIPTOR:
+			switch (ctrl->wValue >> 8) {
+			case USB_DT_DEVICE:
+			case USB_DT_CONFIG:
+			case USB_DT_STRING:
+			case HID_DT_REPORT:
+			case USB_DT_BOS:
+			case USB_DT_HUB:
+			case USB_DT_SS_HUB:
+				return;
+			}
+		}
+		break;
+	case USB_TYPE_CLASS:
+		switch (ctrl->bRequest) {
+		case USB_REQ_GET_INTERFACE:
+		case USB_REQ_GET_CONFIGURATION:
+		case USB_REQ_GET_STATUS:
+		case USB_CDC_GET_NTB_PARAMETERS:
+			return;
+		}
+	}
+	fail("analyze_control_request: unknown control request (0x%x, 0x%x, 0x%x)",
+	     ctrl->bRequestType, ctrl->bRequest, ctrl->wValue);
+}
+#else
+static void analyze_control_request(struct usb_ctrlrequest* ctrl)
```

Comment:
From C reproducer point of view, it would be better to surround the call with #if USB_DEBUG. Otherwise all USB reproducers will have the function call and the empty function, which is confusing.

Reply:
Done